src/renderer/components/+preferences/preferences-navigation/preference-navigation-items.injectable.ts

```diff
@@ -20,6 +20,7 @@ export interface PreferenceNavigationItem {
   isVisible: IComputedValue<boolean>;
   navigate: () => void;
   orderNumber: number;
+  fromExtension?: boolean;
```

Comment:
Note: Ideally, the contract of a `PreferenceNavigationItem` should require no knowledge of extensions. There is a change in the works to make something like this easier to implement, and in a reactive way.

Reply:
Does this work, if an extension is enabled, disabled, and then re-enabled?

Reply:
Most likely this kind of testing functionality (disable/enable extensions after adding to app builder) would require changes in `ApplicationBuilder`. I suggest to provide this test cases on follow up PR.

Reply:
#5609 adds this functionality to `ApplicationBuilder`